woocommerce_fusion/tasks/sync_sales_orders.py

```diff
@@ -551,16 +551,29 @@ def create_or_link_customer_and_address(
 		"""
 		customer_woo_com_email = raw_billing_data.get("email")
 		customer_exists = frappe.get_value("Customer", {"woocommerce_email": customer_woo_com_email})
-		if not customer_exists:
+		address_exists = frappe.get_value("Address", {"address_line1": raw_billing_data.get("address_1"), "state": raw_billing_data.get("state"),
+											"city": raw_billing_data.get('city')})
+		if not customer_exists and not address_exists:
 			# Create Customer
 			customer = frappe.new_doc("Customer")
 			customer_docname = customer_name[:3].upper() + f"{randrange(1, 10**3):03}"
 			customer.name = customer_docname
+
+		elif address_exists:
+			doc = frappe.get_doc("Address", address_exists)
+			for link in doc.links:
+				if link.link_doctype == "Customer":
+					customer = frappe.get_doc("Customer", link.link_name)
+					break
+			if not customer:
+				customer = frappe.new_doc("Customer")
```

Comment:
@AwuyaDaniel this code:
```
customer = frappe.new_doc("Customer")
customer_docname = customer_name[:3].upper() + f"{randrange(1, 10**3):03}"
customer.name = customer_docname
```
is used also in line 557. Why not wrap the code into a new reusable function.?

Reply:
@AwuyaDaniel this too